File: my_solution.js
class TreeNode {
    constructor(value = 0, leftChild = null, rightChild = null) {
        this.value = value
        this.left = leftChild
        this.right = rightChild
    }
}

function sumBTIterative(root) {
    if(!root) return 0
    
    let sum = 0
    let queue = [root]
    while(queue.length){
        const curr = queue.shift()
        sum += curr.value
        if(curr.left) queue.push(curr.left)
        if(curr.right) queue.push(curr.right)
    }
    return sum
}

function sumBTRecursive(root) {
    if(!root) return 0
    
    return root.value + sumBTRecursive(root.left) + sumBTRecursive(root.right) 
}

// Test Cases
console.log(sumBTIterative(null)) // 0
console.log(sumBTIterative(new TreeNode(1, new TreeNode(2), new TreeNode(3)))) // 6 
console.log(sumBTIterative(new TreeNode(1))) // 1
console.log(sumBTRecursive(null)) // 0
console.log(sumBTRecursive(new TreeNode(1, new TreeNode(2), new TreeNode(3)))) // 6
console.log(sumBTRecursive(new TreeNode(1))) // 1

File: my_solution.py
class TreeNode:
    def __init__(self, value = 0, left = None, right = None):
        self.value = value
        self.left = left
        self.right = right

def sumBTIterative(root: TreeNode) -> int:
    if not root:
        return 0
    
    sum = 0
    queue = [root]
    while len(queue):
        curr = queue.pop(0)
        sum += curr.value
        if curr.left:
            queue.append(curr.left)
        if curr.right:
            queue.append(curr.right)
    return sum

def sumBTRecursive(root: TreeNode) -> int:
    if not root:
        return 0
    
    return root.value + sumBTRecursive(root.left) + sumBTRecursive(root.right)

# Test Cases
print(sumBTIterative(None)) # 0
print(sumBTIterative(TreeNode(1, TreeNode(2), TreeNode(3)))) # 6
print(sumBTIterative(TreeNode(1))) # 1
print(sumBTRecursive(None)) # 0
print(sumBTRecursive(TreeNode(1, TreeNode(2), TreeNode(3)))) # 6
print(sumBTRecursive(TreeNode(1))) # 1
